adam-core/src/main/scala/org/bdgenomics/adam/rdd/features/FeatureParser.scala

```diff
@@ -194,58 +230,65 @@ class IntervalListParser extends Serializable {
 
 }
 
+/**
+ * Parser for BED format.
+ *
+ * Specification:
+ * http://www.genome.ucsc.edu/FAQ/FAQformat.html#format1
+ */
 class BEDParser extends FeatureParser {
 
+  def isHeader(line: String): Boolean = {
+    line.startsWith("#") || line.startsWith("browser") || line.startsWith("track")
+  }
+
   override def parse(line: String): Seq[Feature] = {
 
+    if (isHeader(line)) {
+      return Seq()
+    }
+
     val fields = line.split("\t")
     if (fields.length < 3) {
       return Seq()
     }
     val fb = Feature.newBuilder()
-    val cb = Contig.newBuilder()
-    cb.setContigName(fields(0))
-    fb.setContig(cb.build())
-    fb.setFeatureId(UUID.randomUUID().toString)
+    fb.setContigName(fields(0))
 
     // BED files are 0-based space-coordinates, so conversion to
     // our coordinate space should mean that the values are unchanged.
     fb.setStart(fields(1).toLong)
     fb.setEnd(fields(2).toLong)
 
     if (fields.length > 3) {
-      fb.setFeatureType(fields(3))
+      fb.setName(fields(3))
     }
     if (fields.length > 4) {
-      fb.setValue(fields(4) match {
+      fb.setScore(fields(4) match {
         case "." => null
         case _   => fields(4).toDouble
       })
     }
     if (fields.length > 5) {
-      fb.setStrand(fields(5) match {
-        case "+" => Strand.Forward
-        case "-" => Strand.Reverse
-        case _   => Strand.Independent
-      })
+      Features.toStrand(fields(5)).foreach(fb.setStrand(_))
     }
     val attributes = new ArrayBuffer[(String, String)]()
-    if (fields.length > 6) {
+    if (fields.length > 6 && fields(6) != ".") {
```

Comment:
Factor out `fields.length > n && fields(n) != "."` into:

```
def validIdx(n: Int): Boolean = {
  fields.length > n && fields(n) != "."
}
```

Or something like that.